### ISSUE TITLE
Using boundtype variable in type alias without new type syntax gives error

### ISSUE THREAD

**Issue by maltevesper** [microsoft/pyright#6609]
Pyright (pyright 1.1.338) gives a warning when I try to use a typevariable bound to the class to specify a type alias inside. The warning disapears if the new type syntax is used.

```python
class Table[ROWTYPE]:
    type DataType = list[ROWTYPE] # ok

class TableImplicit[ROWTYPE]:
    DataType = list[ROWTYPE] # error
```

> .py:2:17 - error: Generic type alias within class cannot use bound type variables T
> .py:2:5 - error: Generic type alias within class cannot use bound type variables T
> 2 errors, 0 warnings, 0 informations

**Comment by erictraut** [comment]
Thinking about this a bit more, I'm not sure that this should be disallowed for `type` statements. There's a good reason why this wasn't allowed for traditional type aliases (the other two cases in your code sample above) — because they created a new TypeVar scope and implicitly bound type variables found within the definition's type expression. But with a `type` statement, type parameters are explicitly specified, which removes any ambiguity about the intent.

I'm going to open a question in the typing forums to see if anyone can think of a good reason why this shouldn't be supported. If not, I'm going to leave the current behavior as is.

**Comment by erictraut** [comment]
For reference, here's the public discussion in the typing forum: https://discuss.python.org/t/class-scoped-type-statement-that-references-outer-scoped-typevar/40026.

**Comment by maltevesper** [comment]
I am no typing expert (as is probably quite apparent by now), nevertheless I share the sentiment that it should not be an error for PEP695 with type. I added a note over at the python discussion, to summarize here:

I was more taken aback that the typechecker complained when I wanted to use a bound typevar for a non-generic typealias. (I can see how a bound typevar is an issue for a generic typealias).

In short: I currently use the no warning behaviour when I use it with type, and it works the way I want it to: No warnings, and autocompletion works as desired.
I was more confused that the old syntax did not work, and needed your explanation to better understand the reasons for why the old syntax causes errors. (While I have not done enough reading to fully convince myself that the old behaviour is necessary/desireable, I am really happy that the PEP you worked towards gives me the freedom to express what I desire).

Unless someone complains over in the python discussion please keep this behaviour and feel free to close this issue.

**Comment by erictraut** [comment]
> Unless someone complains over in the python discussion please keep this behaviour and feel free to close this issue.

It appears that the general consensus (at least so far) is that this behavior should be allowed for the new `type` statement. I'm therefore going to keep the current behavior and close the issue. If something changes in the other discussion, I'm willing to revisit this, but I think it's unlikley.
